df.world-site.xml

```diff
@@ -58,7 +58,7 @@
 
         <stl-vector name='inhabitants'>
             <pointer>
-                <int16_t/>
+                <int16_t ref-target='historical_figure'/>
                 <int32_t ref-target='historical_figure'/>
```

Comment:
You are right, but all the values I've seen were very low (0, 3, 7, etc) so they would fit in a int16_t.
For the other field anon_2, definitively a int32_t is needed.

Here's a extract of the DF code disassembled with IDA
```C++
          v4_T_Inhabitant@anon_1 = *v3;
          result_ptr = (world_history__figures__endptr - world_history__figures__ptr) >> 2;
          if ( result_ptr )
          {
            if ( v4_T_Inhabitant@anon_1 != -1 )
            {
              v5 = 0;
              v17 = --result_ptr;
              v6 = result_ptr;
              if ( result_ptr >= 0 )
              {
                while ( 1 )                     // binary search
                {
                  result_ptr = (v5 + v6) >> 1;
                  v7_ptr_historical_figure = *(world_history__figures__ptr + 4 * result_ptr);
                  v8_historical_figure@id = *(v7_ptr_historical_figure + 196);

                  // Binary search check
                  if ( v8_historical_figure@id == v4_T_Inhabitant@anon_1 )
                    break;
                  if ( v8_historical_figure@id <= v4_T_Inhabitant@anon_1 )
                    v5 = result_ptr + 1;
                  else
                    v6 = result_ptr - 1;
                  if ( v5 > v6 )
                    goto LABEL_29;
                }
```